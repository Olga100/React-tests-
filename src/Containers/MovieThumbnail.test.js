import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import {MovieThumbnailView} from './MovieThumbnail';


describe('<MovieThumbnailView/>', () => {

    const movie = {
        title: "title",
        posterUrl: "http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg",
        genres: ["horror"],
        actors: ["Vasya"],
        director: "Petya",
        description: "This is a very good movie!",
        stars: 4,
        likes: 42
    };

    it('Renders   component  for the movie list', () => {

        const component = renderer.create(
            <MovieThumbnailView movie={movie}/>
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it('Calls callback on click', () => {
        const selectMovieSpy = jest.fn();
        const content = mount(<MovieThumbnailView movie={movie} selectMovie={selectMovieSpy}/>);

        content.find('.movie-thumbnail-title').simulate('click');

        expect(selectMovieSpy).toHaveBeenCalled();
    });

    it('Check on click for Likes', () => {
        const incrementLikesSpy = jest.fn();
        const decrementLikesSpy = jest.fn();
        const content = shallow(<MovieThumbnailView movie={movie} incrementLikes={incrementLikesSpy}
                                                    decrementLikes={decrementLikesSpy}/>);

        content.find('.thumb').at(0).simulate('click');
        content.find('.thumb').at(1).simulate('click');

        expect(incrementLikesSpy).toHaveBeenCalled();
        expect(decrementLikesSpy).toHaveBeenCalled();
    });

});
