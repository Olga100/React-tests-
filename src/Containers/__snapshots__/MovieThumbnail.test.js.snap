// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieThumbnailView/> Renders   component  for the movie list 1`] = `
<div
  className="movie-thumbnail-title"
  onClick={undefined}
>
  <div
    className="title"
  >
    <small>
      title
    </small>
  </div>
  <div
    className="likes"
  >
    <button
      className="thumb"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-thumbs-o-up"
      />
    </button>
    <button
      className="thumb"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-thumbs-o-down"
      />
    </button>
    <small>
      likes
    </small>
    <hr />
    <span
      className="amountLike"
    >
      42
    </span>
  </div>
  <img
    alt="poster"
    className="image"
    src="http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg"
  />
  <div
    className="clear"
  />
  <div
    className="stars"
  >
    <span
      style={
        Object {
          "cursor": "default",
        }
      }
    >
      <span
        className="star"
        onClick={[Function]}
      >
        ★
      </span>
      <span
        className="star"
        onClick={[Function]}
      >
        ★
      </span>
      <span
        className="star"
        onClick={[Function]}
      >
        ★
      </span>
      <span
        className="star"
        onClick={[Function]}
      >
        ★
      </span>
      <span
        className="star"
        onClick={[Function]}
      >
        ☆
      </span>
    </span>
  </div>
</div>
`;
